enhancements/bare-metal-fulfillment/README.md

```diff
@@ -0,0 +1,360 @@
+
+#### Host Pool Network Update
+
+1. The tenant uses the Fulfillment CLI to update the network configuration of a HostPool.
+2. The O-SAC solution fulfills the request by updating the network configuration of the hosts in the HostPool.
+
+#### Host Pool Expansion
+
+1. The tenant uses the Fulfillment CLI to increase the number of requested hosts specified in a HostPool, updating any desired filters.
+2. The O-SAC solution fulfills the request by performing a reconciliation and adding the requested hosts into the HostPool.
+
+#### Host Pool Reduction
+
+1. The tenant uses the Fulfillment CLI to decrease the number of requested hosts specified in a HostPool. If they wish to remove a specific host, they update their filters to include "not host with name X"; otherwise an arbitrary host will be removed.
+2. The O-SAC solution fulfills the request by performing a reconciliation and removing the requested hosts from the HostPool.
+
+#### Host Operations
+
+1. The tenant uses the Fulfillment CLI to see their Hosts.
+2. The tenant uses the Fulfillment CLI to perform an operation upon the desired Host.
+
+#### Host Pool Deletion
+
+1. The tenant uses the Fulfillment CLI to delete a HostPool.
+2. The O-SAC solution fulfills the request and removes the Host resources corresponding to the hosts in the HostPool.
+
+### API Extensions
+
+#### HostPools
+
+A tenant requests bare metal resources by creating a HostPool with a desired specification. Once created, O-SAC
+will repeatedly attempt to fulfill the request until the state of the HostPool matches its specification. Tenants
+can update a HostPool specification, and the same reconciliation process will perform the needed operations to change
+the state of the HostPool.
+
+In the following example, the tenant is requesting a HostPool with two fc430 hosts. Each host will have `network1` attached
+as a native VLAN on one physical interface; and a trunk port with `network2` as a native VLAN
+and `network3` as a tagged VLAN on a second physical interface.
+
+    apiVersion: o-sac.openshift.io/v1alpha1
+    kind: HostPool
+    metadata:
+      name: example
+    spec:
+      hostRequests:
+      - resourceClass: fc430
+        numberOfHosts: 2
+
+      # The networkAttachments section controls how networks are connected to
+      # physical interfaces.
+      networkAttachments:
+      # This configuration will match any available interface.
+      - primary: network1
+      - primary: network2
+        vlans:
+        - network3
+
+If the tenant wishes to specify host properties, they can filter hosts by specifying hostSelectors; for example, this
+HostPool specification will require that hosts be located on rack R2:
+
+    apiVersion: o-sac.openshift.io/v1alpha1
+    kind: HostPool
+    metadata:
+      name: example
+    spec:
+      hostRequests:
+      - resourceClass: fc430
+        numberOfHosts: 2
+	hostSelectors:
+	- rack: R2
```

Comment:
We probably want to support the standard Kubernetes `matchLabels` and `matchExpression` syntaxes here, so this would look something like:

```suggestion
	hostSelectors:
	  matchLabels:
        row: R2
      matchExpressions:
      - key: cabinet
        op: NotIn
        value: ["C3"]
```

That says, "must be in row R2 but must not be in cabinet C3".